balena/models/device.py

```diff
@@ -1197,3 +1202,95 @@ def get_supervisor_state(self, uuid):
             '/supervisor/v1/device', 'POST', data=data,
             endpoint=self.settings.get('api_endpoint')
         )
+
+    def __check_os_update_target(self, device_info, target_os_version):
+        """
+        """
+
+        if 'uuid' not in device_info and not device_info['uuid']:
+            raise exceptions.OsUpdateError('The uuid of the device is not available')
+
+        if 'is_online' not in device_info and not device_info['is_online']:
+            raise exceptions.OsUpdateError('The device is offline: {uuid}'.format(uuid=uuid))
+
+        if 'os_version' not in device_info and not device_info['os_version']:
+            raise exceptions.OsUpdateError('The current os version of the device is not available: {uuid}'.format(uuid=uuid))
+
+        if 'device_type' not in device_info and not device_info['device_type']:
```

Comment:
If the above is true
```suggestion
        if 'device_type' not in device_info or not device_info['device_type']:
```